### PR TITLE
[2145] Update Makefile

### PR DESCRIPTION
**Pull request by saliceti**
### Context
Make it easy to manage depployments via Makefile

### Changes proposed in this pull request
Use docker image tag from master if not specified
Enforce passcode when running deploy from Makefile
Run terraform commands from the terraform directory
Enable review apps via Makefile

### Guidance to review
- Run deploy-plan
- Successful deployment: https://github.com/DFE-Digital/publish-teacher-training/actions/runs/1008831128
- Successful delete: https://github.com/DFE-Digital/publish-teacher-training/runs/3018051003?check_suite_focus=true

### Checklist

- [ ] Make sure all information from the Trello card is in here
- [ ] Attach to Trello card
- [ ] Rebased master
- [ ] Cleaned commit history
- [ ] Tested by running locally
- [ ] Product Review

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -37,11 +37,11 @@ review:
...
 	$(if $(APP_NAME), , $(error Missing environment variable "APP_NAME", Please specify a name for your review app))
 	$(eval AZ_SUBSCRIPTION=s121-findpostgraduateteachertraining-development)
-	$(eval backend_key=-backend-config=key=$(APP_NAME).terraform.tfstate)
+	$(eval backend_key=-backend-config=key=pr-$(APP_NAME).tfstate)
 	$(eval export TF_VAR_paas_app_environment_config=review)
 	$(eval export TF_VAR_paas_app_environment=review-$(APP_NAME))
-	$(eval export TF_VAR_paas_web_app_host_name-$(APP_NAME))
-	echo https://publish-teacher-training-$(APP_NAME).london.cloudapps.digital will be created in bat-qa space
+	$(eval export TF_VAR_paas_web_app_host_name=$(APP_NAME))
+	echo Review app: https://publish-teacher-training-$(APP_NAME).london.cloudapps.digital in bat-qa space
```
Comment (vigneshmsft) [inline Makefile:44]:
```suggestion
	echo Review app: https://publish-teacher-training-pr-$(APP_NAME).london.cloudapps.digital in bat-qa space
```

File: Makefile [hunk 2]
```diff
@@ -106,26 +110,28 @@ print-app-secrets: install-fetch-config set-azure-account
...
+	cd terraform && . workspace_variables/$(DEPLOY_ENV).sh \
+		&& terraform plan -var-file=workspace_variables/$(DEPLOY_ENV).tfvars
 
 deploy: deploy-init
-	. terraform/workspace_variables/$(DEPLOY_ENV).sh \
-		&& terraform apply -var-file=terraform/workspace_variables/$(DEPLOY_ENV).tfvars -auto-approve terraform
+	cd terraform && . workspace_variables/$(DEPLOY_ENV).sh \
+		&& terraform apply -var-file=workspace_variables/$(DEPLOY_ENV).tfvars -auto-approve
 
 destroy: deploy-init
```
Comment (vigneshmsft) [inline Makefile:132]: deploy-init has terraform apply?
why is this required for destroy?

**Comment by defong** [comment]
[Delete old review apps](https://trello.com/c/Wqg9A0fk/2145-delete-old-review-apps)

File: Makefile [hunk 3]
```diff
@@ -106,26 +110,27 @@ print-app-secrets: install-fetch-config set-azure-account
...
-	. terraform/workspace_variables/$(DEPLOY_ENV).sh \
-		&& terraform destroy -var-file=terraform/workspace_variables/$(DEPLOY_ENV).tfvars terraform
+	cd terraform && . workspace_variables/$(DEPLOY_ENV).sh \
+		&& terraform destroy -var-file=workspace_variables/$(DEPLOY_ENV).tfvars -auto-approve
```
Comment (vigneshmsft) [inline Makefile:133]: can we add `-auto-approve` only for the ci command?

**Review by vigneshmsft — APPROVED** [review]
can we add `-auto-approve` in destroy only for the ci command?

File: Makefile (hunk 3 above)
Comment (saliceti) [reply Makefile]: Good idea! Added

**Comment by vigneshmsft** [comment]
[Use make commands in GitHub workflows](https://trello.com/c/2yLKsnsh/2199-use-make-commands-in-github-workflows)
